bbq/vm/test/vm_test.go

```diff
@@ -5450,3 +5450,23 @@ func TestFunctionExpression(t *testing.T) {
 	assert.Equal(t, interpreter.NewUnmeteredIntValueFromInt64(6), actual)
 
 }
+
+func TestInnerFunction(t *testing.T) {
+
+	t.Parallel()
+
+	actual, err := compileAndInvoke(t,
+		`
+          fun test(): Int {
+              fun addOne(_ x: Int): Int {
+                  return x + 1
```

Comment:
Maybe also add some test(s) with pre/post conditions + conditional returns. I think it should seamlessly work.
If not, we can leave it for now - I'll have a look why it's not working.

Reply:
It looks like both function expressions and inner functions are not desugared yet. Testing both triggers:
```go
	// Function conditions must have been desugared to statements.
	// So there shouldn't be any condition at this point.
	if functionBlock.PreConditions != nil ||
		functionBlock.PostConditions != nil {
		panic(errors.NewUnreachableError())
	}
```